### ISSUE TITLE
Problem with cabal-3.10 and haddock: internal error when calculating transitive package dependencies

### ISSUE THREAD

**Issue by maksbotan** [haskell-CI/haskell-ci#658]
Hi!

This issue is not with `haskell-ci` itself, but I figured that you might have the right expertise to give a helpful insight into this problem.

Recently I received a PR in openapi3 repo: https://github.com/biocad/openapi3/pull/82. Among other things, the author updated `haskell-ci` to 0.16 and added GHC 9.6.1 build. With this configuration the build started failing on all versions lower than 9.4 at Haddock stage:

https://github.com/biocad/openapi3/actions/runs/4708381042

```
setup: internal error when calculating transitive package dependencies.
Debug info: []

Error: cabal-3.10.1.0: Failed to build documentation for openapi3-3.2.3.
```

I believe this is related to https://github.com/haskell/cabal/issues/5423, however in that thread it is mentioned that the issue is potentially fixed in cabal 3.8.

I remember that you had some troubles with cabal and haddock before releasing `haskell-ci-0.16`, do you know anything about this one?

Thanks in advance.

**Comment by phadej** [comment]
I haven't experienced any issues with `cabal-install-3.10.1.0`. Please ask on the `cabal-install`'s issue tracker.

**Comment by ysangkok** [comment]
@maksbotan I think the openapi3 issue is distinct from https://github.com/haskell/cabal/issues/5423, since the Cabal issue also affects Cabal 3.6.2.0 and 3.8.1.0, while the openapi3 issue is resolved by reverting to 3.6.2.0.

It doesn't seem like haskell-ci provides the option of using different Cabal versions for different GHC versions. But it seems easy enough to patch this in the generated script, so this feature may not be necessary.

**Comment by phadej** [comment]
I tried locally, and culprit seems to be `--haddock-all`. You can configure to haddock only library part with `haddock-components: libs`.
